packages/realm/src/schema/property_schema_grammar.txt

```diff
@@ -0,0 +1,114 @@
+---------------------------------------------------------------------
+GRAMMAR (from Proposal 2):
+---------------------------------------------------------------------
+- Defines the property schema a user can provide
+---------------------------------------------------------------------
+
+
+----------------
+GRAMMAR NOTATION
+----------------
+
+<rule name>:<rule>      Rule structure
+<single quoted string>  Exact literal
+<double quoted string>  Exact string literal
+<uppercase word>        Varying literal
+<lowercase word>        A call to another rule
+
+:                       Start of rule 
+;                       End of rule
+x*                      Match zero or more of x
+x+                      Match one or more of x
+x?                      Match zero or one of x
+x1 x2                   Match x1 then x2
+x1 | x2                 Match x1 or x2
+( x )                   Match x (everything inside the parentheses)
+< ... >                 Description (...) of a rule
+//                      Comment with no semantic meaning
+
+
+----------------
+START RULE
+----------------
+
+propertySchema:
+    propertySchemaString
+    | propertySchemaObject
+    ;
+
+
+----------------
+OTHER RULES
+----------------
+
+primitiveType:
+    "bool"
+    | "int"
+    | "float"
+    | "double"
+    | "decimal128"
+    | "objectId"
+    | "string"
+    | "data"
+    | "date"
+    | "mixed"
+    | "uuid"
+    ;
+
+collectionType:
+    "list"
+    | "dictionary"
+    | "set"
+    ;
+
+relationshipType:
+    "object"
+    | "linkingObjects"
+    ;
+
+propertySchemaString:
+    ( primitiveType | USER_TEXT ) ( "?" )? ( "[]" | "{}" | "<>" )?
+    ;
+
+propertySchemaObject:
+    '{' propertySchemaObjectFields '}'
+    ;
```

Comment:
Yeah we don't care about the surrounding JS syntax for how to represent an object, but wasn't completely sure how to write a grammar without that. I see your suggestions for writing that part in TS so that seems good 👍